test/run_clippy.sh

```diff
@@ -0,0 +1,138 @@
+               "clippy::blacklisted_name" \
+               "clippy::block_in_if_condition_stmt" \
+               "clippy::bool_comparison" \
+               "clippy::cast_lossless" \
+               "clippy::clone_on_copy" \
+               "clippy::cmp_owned" \
+               "clippy::collapsible_if" \
+               "clippy::const_static_lifetime" \
+               "clippy::correctness" \
+               "clippy::deref_addrof" \
+               "clippy::expect_fun_call" \
+               "clippy::for_kv_map" \
+               "clippy::get_unwrap" \
+               "clippy::identity_conversion" \
+               "clippy::if_let_some_result" \
+               "clippy::len_zero" \
+               "clippy::let_and_return" \
+               "clippy::let_unit_value" \
+               "clippy::map_clone" \
+               "clippy::match_bool" \
+               "clippy::match_ref_pats" \
+               "clippy::needless_bool" \
+               "clippy::needless_collect" \
+               "clippy::needless_range_loop" \
+               "clippy::ok_expect" \
+               "clippy::op_ref" \
+               "clippy::option_map_unit_fn" \
+               "clippy::or_fun_call" \
+               "clippy::println_empty_string" \
+               "clippy::ptr_arg" \
+               "clippy::redundant_closure" \
+               "clippy::redundant_pattern_matching" \
+               "clippy::single_char_pattern" \
+               "clippy::single_match" \
+               "clippy::string_lit_as_bytes" \
+               "clippy::toplevel_ref_arg" \
+               "clippy::unit_arg" \
+               "clippy::unnecessary_operation" \
+               "clippy::unreadable_literal" \
+               "clippy::unused_label" \
+               "clippy::unused_unit" \
+               "clippy::useless_asref" \
+               "clippy::useless_format" \
+               "clippy::useless_let_if_seq" \
+               "clippy::useless_vec" \
+               "clippy::write_with_newline" \
+               "renamed_and_removed_lints")
+
+ # stick together unexamined lints
+for unexamined_lint in "${unexamined_lints[@]}"
+do
+  [[ -n $unexamined_lint ]] && rules_string+="-A ${unexamined_lint} "
+done
+
+ # stick together allowed lints
+for allowed_lint in "${allowed_lints[@]}"
+do
+  [[ -n $allowed_lint ]] && rules_string+="-A ${allowed_lint} "
+done
+
+ # stick together lints to fix
+for lint_to_fix in "${lints_to_fix[@]}"
+do
+  [[ -n $lint_to_fix ]] && rules_string+="-W ${lint_to_fix} "
+done
+
+ # stick together denied lints
+for denied_lint in "${denied_lints[@]}"
+do
+  [[ -n $denied_lint ]] && rules_string+="-D ${denied_lint} "
+done
+
+ component=${1?component argument required}
+echo "--- Running clippy!"
+echo "Clippy rules: cargo clippy --all-targets --tests -- ${rules_string}"
+cd "components/$component"
+# If we include double quotes in this string, the invocation of cargo breaks
+# shellcheck disable=SC2086
+cargo clippy --all-targets --tests -- ${rules_string} 
```

Comment:
The issue here is that quoting `${rules_string}` tells the shell: this is 1 argument, and `clippy` understandably doesn't know what to do with a single argument like `"-A clippy::module_inception … -D renamed_and_removed_lints"`. What we want here is not to build up a rules string, but an argument *array*. Then we can appropriately quote:
```suggestion
cargo clippy --all-targets --tests -- "${clippy_args[@]}"
```

Reply:
Arg! I knew it was something simple I was messing up.

Reply:
Arrays in bash are very useful, but I wouldn't call them "simple" 😄